components/nexus-blobstore-file/src/main/java/org/sonatype/nexus/blobstore/file/MapdbBlobMetadataStore.java

```diff
@@ -109,7 +111,9 @@ protected void doStop() throws Exception {
   }
 
   private HTreeMap<BlobId, MetadataRecord> entries(final DB db) {
-    return db.getHashMap("entries");
+    return db.createHashMap("entries")
+        .counterEnable()
```

Comment:
this will have some impact on insert/update but otherwise ascertaining the size requires iterating over the entire collection